pkg/build/strategies/sti/sti.go

```diff
@@ -215,13 +215,18 @@ func (builder *STI) Build(config *api.Config) (*api.Result, error) {
 		}
 	}
 
-	if len(config.AssembleUser) > 0 {
-		glog.V(1).Infof("Running %q in %q as %q user", api.Assemble, config.Tag, config.AssembleUser)
+	assembleUser, err := builder.docker.GetAssembleUser(config.AssembleUser, config.BuilderImage)
+	if err != nil {
+		return builder.result, err
+	}
+
+	if len(assembleUser) > 0 {
+		glog.V(1).Infof("Running %q in %q as %q user", api.Assemble, config.Tag, assembleUser)
 	} else {
 		glog.V(1).Infof("Running %q in %q", api.Assemble, config.Tag)
 	}
 	startTime := time.Now()
-	if err := builder.scripts.Execute(api.Assemble, config.AssembleUser, config); err != nil {
+	if err := builder.scripts.Execute(api.Assemble, assembleUser, config); err != nil {
```

Comment:
@bparees Thanks!

I'm moving this to strategies.go, but while looking to this check I couldn't find where, in pullAndCheck function and also in CheckAllowedUser function the config.AssembleUser is verified.

Am I missing something here?

Thanks!!!